pimd/pim_neighbor.c

```diff
@@ -419,8 +419,8 @@ void pim_neighbor_free(struct pim_neighbor *neigh)
 	list_delete(&neigh->upstream_jp_agg);
 	THREAD_OFF(neigh->jp_timer);
 
-	if (neigh->bfd_info)
-		pim_bfd_info_free(&neigh->bfd_info);
+	if (neigh->bfd_session)
+		bfd_sess_free(&neigh->bfd_session);
```

Comment:
`pim_neighbor_free` is only called from `pim_neighbor_delete` which already frees the BFD session.

Reply:
I kept it here and removed from `pim_neighbor_delete` because the former might be called as a callback by the list deletion function.